### PR TITLE
properly handle object creation error in template instantiate

### PR DESCRIPTION
**Pull request by bparees**
https://bugzilla.redhat.com/show_bug.cgi?id=1661845

### CODE REVIEW

**Comment by bparees** [comment]
@smarterclayton fyi

File: pkg/template/controller/templateinstance_controller.go [hunk 1]
```diff
@@ -501,7 +501,7 @@ func (c *TemplateInstanceController) instantiate(templateInstance *templatev1.Te
 			}
 		}
 		if createErr != nil {
-			allErrors = append(allErrors, mappingErr)
+			allErrors = append(allErrors, createErr)
```
Comment (coreydaley) [inline pkg/template/controller/templateinstance_controller.go]: must have been a copy and paste from above ...

File: test/extended/templates/templateinstance_fail.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			err = cli.Run("create").Args("-f", templatefixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			template, err = cli.TemplateClient().TemplateV1().Templates(cli.Namespace()).Get("invalidtemplateinstance", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.It("should report report a failure on creation", func() {
```
Comment (coreydaley) [inline test/extended/templates/templateinstance_fail.go]: s/report report/report/

File: pkg/template/controller/templateinstance_controller.go (hunk 1 above)
Comment (bparees) [reply pkg/template/controller/templateinstance_controller.go]: yeah, seems like.

**Comment by bparees** [comment]
missed a file save. updated.

File: test/extended/templates/templateinstance_fail.go (hunk 2 above)
Comment (bparees) [reply test/extended/templates/templateinstance_fail.go]: fixed

File: test/extended/templates/templateinstance_fail.go [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+			err = cli.Run("create").Args("-f", templatefixture).Execute()
+			o.Expect(err).NotTo(o.HaveOccurred())
+
+			template, err = cli.TemplateClient().TemplateV1().Templates(cli.Namespace()).Get("invalidtemplateinstance", metav1.GetOptions{})
+			o.Expect(err).NotTo(o.HaveOccurred())
+		})
+
+		g.It("should report a failure on creation", func() {
+			var err error
```
Comment (coreydaley) [inline test/extended/templates/templateinstance_fail.go]: Just a question here: Why use var err error here?

Comment (bparees) [reply test/extended/templates/templateinstance_fail.go]: was copied from another test, might have made more sense the way the other test was structured.

anyway i had to redo the test a bit so now that's gone.

**Comment by bparees** [comment]
the new test passed in e2e-gcp so we can let the bot take it from here.

**Comment by bparees** [comment]
/skip

Ben Parees | OpenShift

**Comment by bparees** [comment]
It's a Christmas miracle.


Ben Parees | OpenShift

**Pull request merged**